public/js/cin.js

```diff
@@ -0,0 +1,527 @@
+          placeholder="Search"
+          aria-label="Search"
+          onKeyUp=${handleSearch}
+          style="width: 25em;  max-width: 90%;"
+        />`,
+      ];
+      interviews.forEach((event) => {
+        // Spaces seem to break the search, so just yeet the space characters
+        if (
+          searchTerm.current == "" ||
+          event.interview.position.full_name
+            .toLowerCase()
+            .replace(/\s/g, "")
+            .search(
+              escapeRegExp(searchTerm.current).toLowerCase().replace(/\s/g, "")
+            ) != -1 ||
+          prettifyCandidates(event.interview.candidates)
+            .toLowerCase()
+            .replace(/\s/g, "")
+            .search(
+              escapeRegExp(searchTerm.current).toLowerCase().replace(/\s/g, "")
+            ) != -1
+        ) {
+          if (new Date(event.end_time).getTime() < Date.now()) {
+            var youtube = event.interview.youtube_id;
+            tmp.push(html`<${PastScheduleCard}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+              youtubeStatus=${youtube != null
+                ? html`<a class="cin-text-2" href="javascript:void(0)"
+                    >Watch on YouTube</a
+                  >`
+                : "Available on YouTube Soon"}
+              callback=${updateYoutube}
+              youtubeID=${youtube}
+            />`);
+          } else if (new Date(event.start_time).getTime() > Date.now()) {
+            let time =
+              new Date(event.start_time).toLocaleTimeString().slice(0, -3) +
+              " - " +
+              new Date(event.end_time).toLocaleTimeString().slice(0, -3);
+            tmp.push(html`<${FutureScheduleCard}
+              time=${time}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+            />`);
+          } else {
+            let time =
+              "Now - " +
+              new Date(event.end_time).toLocaleTimeString().slice(0, -3);
+            tmp.push(html`<${LiveScheduleCard}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+              time=${time}
+            />`);
+          }
+        }
+      });
+      if (tmp.length == 1) {
+        // No Interviews, Only Search
+        if (searchTerm.current == "") {
+          setSlots([html`<h2 class="text-center">Coming Soon...</h2>`]);
+        } else {
+          setSlots(
+            tmp.concat([
+              html`<br />
+                <h2 class="text-center">No Results</h2>`,
+            ])
+          );
+        }
+      } else {
+        setSlots(tmp);
```

Comment:
Oh sweet merciful fuck, please don't put ReactNodes (or the preact equivalent) in state. That's literally the antithesis of how React/Preact is meant to work - store the _data_ in the state, then build it up from it in the render function. If it's too complicated, extract it out into a subcomponent.